src/modules/orders/orders.service.ts

```diff
@@ -0,0 +1,193 @@
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
+
+import { Errors } from 'src/common/errors';
+import { Cart } from 'src/modules/cart/cart.entity';
+import { OrderResponseDTO } from 'src/modules/orders/dto/order-response.dto';
+import { Order } from 'src/modules/orders/entities/order.entity';
+import { ProductResponseDTO } from 'src/modules/products/dto/product-response.dto';
+import { User } from 'src/modules/users/user.entity';
+
+import { Product } from '../products/entities/product.entity';
+
+import { OrderDTO } from './dto/order.dto';
+import { BuyerOrder } from './entities/buyer-order.entity';
+import { Status } from './entities/order-status.enum';
+
+@Injectable()
+export class OrdersService {
+  constructor(
+    @InjectRepository(Order)
+    private readonly orderRepository: Repository<Order>,
+    @InjectRepository(User)
+    private readonly userRepository: Repository<User>,
+    @InjectRepository(Product)
+    private readonly productRepository: Repository<ProductResponseDTO>,
+    @InjectRepository(BuyerOrder)
+    private readonly buyerOrderRepository: Repository<BuyerOrder>,
+    @InjectRepository(Cart)
+    private readonly cartRepository: Repository<Cart>,
+  ) {}
+
+  async findByVendor(vendorId: string): Promise<OrderResponseDTO[]> {
+    try {
+      const vendor = await this.userRepository.findOne({
+        where: { id: vendorId },
+      });
+
+      if (!vendor) {
+        throw new NotFoundException(Errors.USER_NOT_FOUND);
+      }
+
+      const orders = await this.orderRepository.find({
+        where: { vendorId: vendorId },
+        relations: ['products'],
+        order: { createdAt: 'DESC' },
+      });
+
+      if (!orders.length) {
+        throw new NotFoundException(Errors.ORDERS_NOT_FOUND);
+      }
+
+      return orders.map((order) => new OrderResponseDTO(order));
+    } catch (error) {
+      if (error instanceof NotFoundException) {
+        throw error;
+      }
+      throw new InternalServerErrorException(
+        Errors.FAILED_TO_FETCH_ORDERS_BY_VENDOR,
+      );
+    }
+  }
+
+  async createOrdersForUser(
+    userId: string,
+    shippingPrice: number,
+  ): Promise<void> {
+    try {
+      const user = await this.userRepository.findOne({ where: { id: userId } });
+
+      if (!user) {
+        throw new NotFoundException(Errors.USER_NOT_FOUND);
+      }
+
+      const cartItems = await this.cartRepository.find({
+        where: { userId: userId },
+      });
+
+      if (!cartItems.length) {
+        throw new NotFoundException(Errors.CART_IS_EMPTY);
+      }
+
+      const ordersMap = new Map<string, OrderDTO>();
+
+      for (const cartItem of cartItems) {
+        const vendor = await this.userRepository.findOne({
+          where: { id: cartItem.vendorId },
+        });
+
+        if (!vendor) {
+          throw new NotFoundException(Errors.VENDOR_NOT_FOUND);
+        }
+
+        const productResponseDTO = await this.productRepository.findOne({
```

Comment:
why it called DTO? DTO = data transfer object

Reply:
fixed